components/brave_ads/core/internal/common/url/url_util_internal.cc

```diff
@@ -0,0 +1,210 @@
+  }
+
+  // We can reject everything else!
+  return false;
+}
+
+std::optional<size_t> GetRegistryLength(const GURL& url) {
+  // `EXCLUDE_PRIVATE_REGISTRIES` is intentional because we want to return
+  // only the registry, not the private registry.
+  const size_t registry_length =
+      net::registry_controlled_domains::GetRegistryLength(
+          url, net::registry_controlled_domains::EXCLUDE_UNKNOWN_REGISTRIES,
+          net::registry_controlled_domains::EXCLUDE_PRIVATE_REGISTRIES);
+  if (registry_length == std::string::npos || registry_length == 0) {
+    return std::nullopt;  // No registry.
+  }
+
+  return registry_length;
+}
+
+std::optional<std::string> GetDomain(const GURL& url) {
+  if (!url.is_valid() || !url.has_host() || url.HostIsIPAddress()) {
+    return std::nullopt;  // No registry.
+  }
+
+  const std::optional<size_t> registry_length = GetRegistryLength(url);
+  if (!registry_length) {
+    return std::nullopt;  // No registry.
+  }
+
+  const std::string host = url.host();
+
+  // Remove the registry from the host.
+  const size_t domain_length =
+      host.length() - (*registry_length + /*'.' before registry*/ 1);
+  CHECK_GE(host.length(), domain_length);
+  return host.substr(0, domain_length);
+}
+
+std::optional<std::string> GetRegistryControlledDomain(const GURL& url) {
+  if (!url.is_valid() || !url.has_host() || url.HostIsIPAddress()) {
+    return std::nullopt;  // No registry.
+  }
+
+  const std::string_view host_piece = url.host_piece();
+
+  if (!net::registry_controlled_domains::HostHasRegistryControlledDomain(
+          host_piece,
+          net::registry_controlled_domains::EXCLUDE_UNKNOWN_REGISTRIES,
+          net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES)) {
+    return std::nullopt;
+  }
+
+  const std::string domain_and_registry =
+      net::registry_controlled_domains::GetDomainAndRegistry(
+          host_piece,
+          net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
+  if (domain_and_registry.empty()) {
+    return std::nullopt;  // No domain.
+  }
+
+  const std::optional<size_t> registry_length = GetRegistryLength(url);
+  if (!registry_length) {
+    return std::nullopt;  // No registry.
+  }
+
+  // Remove the registry from the host.
+  const size_t domain_length = domain_and_registry.length() -
+                               (*registry_length + /*'.' before registry*/ 1);
+  CHECK_GE(domain_and_registry.length(), *registry_length);
+  return domain_and_registry.substr(0, domain_length);
+}
+
+bool IsHostOnPublicSuffixList(const GURL& url) {
+  if (!url.is_valid() || !url.has_host()) {
+    return false;
+  }
+
+  const std::optional<std::string> domain = GetDomain(url);
+  if (!domain) {
+    return false;
+  }
+
+  const std::optional<std::string> registry_controlled_domain =
+      GetRegistryControlledDomain(url);
+  if (!registry_controlled_domain) {
+    return false;
+  }
+
+  const bool registry_controlled_domain_has_private_registries =
```

Comment:
technically i think `co.uk` is considered a public registry even though it has a dot

Reply:
Working with aseren to rename this variable. But the tests do pass.

Reply:
Renamed variable, and will wait for follow up based on other comments. Thanks

Reply:
This code has now been removed after further discussions around example.mysite.appspot.com, thanks